Feature/clean database command.php

demichl68 opened the pull request:
Adds db cleanup command.
Unifies Jira config.
Introduces enhanced jira ticket id from branch name extraction methods when using git branch prefixes such as `feature/foo`, `bugfix/foo`, `hotfix/foo` or `release/foo`.

Solves https://github.com/icanhazstring/duck-pony/issues/5

Commit:
# Provides Command to remove orphaned databases

Commit:
# Prodivdes example configuration.

Commit:
# Minor improvements

Commit:
# Minor improvements

Commit:
# Fixes Author Tag

Commit:
# Fixes output.

Commit:
# Reverts output changes.

Commit:
Merge remote-tracking branch 'forkorigin/master' into feature/CleanDatabaseCommand.php

# Conflicts:
#	composer.json
#	composer.lock
#	src/Console/Command/CleanBranchCommand.php

Commit:
- Unifies Jira config.
- Adds DB Clean command to console application.

Commit:
Merge remote-tracking branch 'forkorigin/master' into feature/CleanDatabaseCommand.php

# Conflicts:
#	src/Console/Command/CleanBranchCommand.php

Commit:
- Resolves merge conflicts.

Commit:
- Updates Changelog and Readme.

Commit:
- Adds tempa substitutes to Readme

Commit:
- Removes return typehint.

Commit:
- Explicitly casts index to int.

demichl68 commented:
The changes are breaking.

icanhazstring commented on config/config.yml.dist:
```diff
@@ -1,8 +1,15 @@
+  password: ${db_password}
+
+Jira:
   hostname: ${host}
```
Can you rename this as `jira_host` as well?
So if you run `vendor/bin/tempa` the key for replacement is easier to distinquish

icanhazstring commented on src/Console/Command/CleanBranchCommand.php:
```diff
@@ -19,10 +19,7 @@
-     * @inheritdoc
-     */
-    protected function configure(): void
+    protected function configure()
```
Can you re add the return type `:void`

icanhazstring commented on src/Console/Command/CleanBranchCommand.php:
```diff
@@ -59,15 +57,16 @@ protected function execute(InputInterface $input, OutputInterface $output): int
-                'jiraHost'     => $config['CleanBranch']['hostname'],
-                'jiraUser'     => $config['CleanBranch']['username'],
-                'jiraPassword' => $config['CleanBranch']['password']
+                'jiraHost'     => $config['Jira']['hostname'],
```
Same here for changed config values

demichl68 replied:
I changed the .dist values for convenient usage within tempa.
What do you want me to change here exactly?

icanhazstring replied:
Simply the keys itself like for mysql => `mysql_hostname`
So we don't have `hostname` to be used with `tempa` but instead `jira_hostname`

---
Edit: Exactly this :)
Edit2: Nevermind, me stupid, everything good

icanhazstring commented on src/Console/Command/CleanBranchCommand.php:
```diff
@@ -137,7 +138,7 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             }
 
-            $progressBar->setProgress($index);
+            $progressBar->setProgress((int) $index);
```
This might not work. Since `$index` here is always the path of the `$dir` itself.
So you might want to replace this with `$progressBar->advance()`

icanhazstring commented on src/Console/Command/CleanMySQLDatabaseCommand.php:
```diff
@@ -0,0 +1,177 @@
+            $config['MySQL']['password'],
+            $config['MySQL']['hostname']
+        )) {
+            throw new \RuntimeException('MySQL config is incorrect! Username password and hostname required!');
```
There might be a `,` (comma)  missing between Username and password

icanhazstring commented on src/Console/Command/CleanMySQLDatabaseCommand.php:
```diff
@@ -0,0 +1,177 @@
+
+        try {
+            $issueService = new IssueService(new ArrayConfiguration([
+                'jiraHost'     => $config['Jira']['hostname'],
```
As well here, change jira config values

icanhazstring commented on src/Console/Command/CleanMySQLDatabaseCommand.php:
```diff
@@ -0,0 +1,177 @@
+     *
+     * @return void
+     */
+    protected function configure()
```
Make sure to use the `: void` returntype

icanhazstring commented on src/Console/Command/CleanMySQLDatabaseCommand.php:
```diff
@@ -0,0 +1,177 @@
+
+        if (empty($remove)) {
+            $io->title('Found no databases to remove!');
+            return;
```
Maybe return `0` here since the script has a "clean" exit state.

icanhazstring commented on src/Console/Command/CleanMySQLDatabaseCommand.php:
```diff
@@ -0,0 +1,177 @@
+     * @return int|null|void
+     * @throws \JiraRestApi\JiraException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
```
Since you only  return `int` (if you change the one `null` return) you can have `: int` return type here

icanhazstring reviewed (changes requested):
Thanks for the PR 👍 
Just some minor adjustments.

Since you changed the configuration, can you please provide a `MIGRATION.md` so people know how to migrate from `0.2` to `0.3`.
Thanks ❤️ 
E.g.

```
# Migrations

## 0.2 to 0.3
Adjust config.yml .....
```

demichl68 commented:
Thanks @icanhazstring 
Will do asap :)

Commit:
- Fixes return types.
- Fixes console application returns.
- Changes Jira config.dist values
- Adds MIGRATION.md

icanhazstring commented:
Merged and thanks 🎉
